compose/config/config.py

```diff
@@ -835,6 +836,21 @@ def finalize_service_volumes(service_dict, environment):
                 finalized_volumes.append(MountSpec.parse(v, normalize, win_host))
             else:
                 finalized_volumes.append(VolumeSpec.parse(v, normalize, win_host))
+
+        duplicate_mounts = []
+        mounts = [(v.external, v.internal) for v in [_v.as_volume_spec()
+                                                     if hasattr(_v, 'as_volume_spec')
+                                                     else _v for _v in finalized_volumes]]
+        for external, internal in mounts:
+            if external and list(map(itemgetter(0), mounts)).count(
+                    external) > 1 or list(map(itemgetter(1), mounts)).count(internal) > 1:
+                duplicate_mounts.append("%s:%s" % (external, internal))
```

Comment:
We're only concerned about `v.internal` duplicates - Mounting the same folder or volume in several (distinct) places inside the container is completely valid. As such, I would rewrite it thus:
```python
duplicate_mounts = []
mounts = [v.as_volume_spec() if isinstance(v, MountSpec) else v for v in finalized_volumes]
for m in mounts:
  if list(map(attrgetter('internal', mounts)).count(m.internal) > 1:
    duplicate_mounts.append(m.repr())
```